lib/galaxy/model/store/__init__.py

```diff
@@ -2215,6 +2217,14 @@ def add_dataset_collection(
     def add_dataset(self, dataset: model.DatasetInstance, include_files: bool = True) -> None:
         self.included_datasets[dataset] = (dataset, include_files)
 
+    def _ensure_dataset_file_exists(self, dataset: model.DatasetInstance) -> None:
+        state = dataset.dataset.state
+        if state in [model.Dataset.states.OK] and not dataset.file_name:
+            log.error(
+                f"Dataset [{dataset.id}] does not exists on on object store [{dataset.dataset.object_store_id or 'None'}], while trying to export."
+            )
+            raise Exception(f"Cannot export dataset: {dataset.name}")
```

Comment:
Do these messages appear in the task error message ? If yes, it might be good to include the encoded dataset id in the message. If the exception message is not user visible we should include the decoded dataset id here. I think having that id will enable admins to more easily debug why they're not available.

For users it might also be a good idea to include the hid. Maybe something like

```suggestion
            raise Exception(f"Cannot export history dataset {dataset.hid}: {dataset.name} with id {self.exported_key(dataset.id)}")
```